### ISSUE TITLE
No entry for `new Date(...)` in lib/core.js

### ISSUE THREAD

**Issue by bolinfest** [facebook/flow#188]
Reading through https://github.com/facebook/flow/blob/master/lib/core.js, there is no `constructor` entry in `declare class Date {`.

**Comment by josephsavona** [comment]
@avikchaudhuri I confirmed that `var date: Date = new Date()` passes. Looks like this issue can be closed?

**Comment by bolinfest** [comment]
@josephsavona I do not believe that is a sufficient test. `Date` has a very complicated constructor: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date.

**Comment by josephsavona** [comment]
@bolinfest Flow accepts all of the below cases. Do you have an example that is valid JS but that Flow rejects?

```
var date1: Date = new Date();
var date2: Date = new Date('2015/01/01');
var date3: Date = new Date(2015, 01, 01);
var date4: Date = new Date(1);
```

**Comment by bolinfest** [comment]
@josephsavona I don't know if that proves anything. I think Flow ignores extra arguments, which is why it doesn't flag anything you are seeing. The real test is whether it rejects invalid arguments. For example, if I write:

    new Date({}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {});

I don't see any errors, but I should.

**Comment by knsn1994** [comment]
@bolinfest you are right. Flow ignores extra arguments. So we cannot really Flag 
    new Date({}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {});

**Comment by jjergus** [comment]
We can flag new Date({}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}); because {} doesn't match the required type of the arguments (string or int for the first argument, int for others). #539 takes care of this.

Ignoring extra arguments is probably OK because JavaScript ignores them as well. Extra arguments make your code ugly, but won't cause actual bugs.

One thing we might want to flag though are invalid argument combinations like new Date(string, int). This is not solved by #539 because Flow doesn't supported overloaded constructor definitions yet.

**Comment by samwgoldman** [comment]
This can be closed, now that #539 is merged.

**Comment by samwgoldman** [comment]
`Date` constructor is was added to flowlib in 0.13.
